synapse/app/homeserver.py

```diff
@@ -512,6 +522,17 @@ def generate_user_daily_visit_stats():
     # table will decrease
     clock.looping_call(generate_user_daily_visit_stats, 5 * 60 * 1000)
 
+    def generate_monthly_active_users():
+        count = 0
+        if hs.config.limit_usage_by_mau:
+            count = hs.get_datastore().count_monthly_users()
+        current_mau_gauge.set(float(count))
+        max_mau_value_gauge.set(float(hs.config.max_mau_value))
+        limit_usage_by_mau_gauge.set(float(hs.config.limit_usage_by_mau))
+
+    generate_monthly_active_users()
+    clock.looping_call(generate_monthly_active_users, 5 * 60 * 1000)
```

Comment:
Does it make sense for this to be done if `hs.config.limit_usage_by_mau` is false?

Reply:
right yes, need to be called once but the looping is a waste